packages/components/src/tab-panel/index.js

```diff
@@ -29,87 +29,71 @@ const TabButton = ( { tabId, onClick, children, selected, ...rest } ) => (
+	const [ selected, setSelected ] = useState(
+		initialTabName || ( tabs.length > 0 ? tabs[ 0 ].name : null )
+	);
 
-		this.handleClick = this.handleClick.bind( this );
-		this.onNavigate = this.onNavigate.bind( this );
-
-		this.state = {
-			selected:
-				initialTabName || ( tabs.length > 0 ? tabs[ 0 ].name : null ),
-		};
-	}
-
-	handleClick( tabKey ) {
-		const { onSelect = noop } = this.props;
-		this.setState( {
-			selected: tabKey,
-		} );
+	const handleClick = ( tabKey ) => {
+		setSelected( tabKey );
 		onSelect( tabKey );
-	}
+	};
 
-	onNavigate( childIndex, child ) {
+	const onNavigate = ( childIndex, child ) => {
 		child.click();
-	}
-
-	render() {
-		const { selected } = this.state;
-		const {
-			activeClass = 'is-active',
-			className,
-			instanceId,
-			orientation = 'horizontal',
-			tabs,
-		} = this.props;
+	};
 
-		const selectedTab = find( tabs, { name: selected } );
-		const selectedId = instanceId + '-' + selectedTab.name;
+	const selectedTab = find( tabs, { name: selected } );
+	const selectedId = instanceId + '-' + selectedTab.name;
 
-		return (
-			<div className={ className }>
-				<NavigableMenu
-					role="tablist"
-					orientation={ orientation }
-					onNavigate={ this.onNavigate }
-					className="components-tab-panel__tabs"
-				>
-					{ tabs.map( ( tab ) => (
-						<TabButton
-							className={ classnames(
-								'components-tab-panel__tabs-item',
-								tab.className,
-								{
-									[ activeClass ]: tab.name === selected,
-								}
-							) }
-							tabId={ instanceId + '-' + tab.name }
-							aria-controls={
-								instanceId + '-' + tab.name + '-view'
+	return (
+		<div className={ className }>
+			<NavigableMenu
+				role="tablist"
+				orientation={ orientation }
+				onNavigate={ onNavigate }
+				className="components-tab-panel__tabs"
+			>
+				{ tabs.map( ( tab ) => (
+					<TabButton
+						className={ classnames(
+							'components-tab-panel__tabs-item',
+							tab.className,
+							{
+								[ activeClass ]: tab.name === selected,
 							}
-							selected={ tab.name === selected }
-							key={ tab.name }
-							onClick={ partial( this.handleClick, tab.name ) }
-						>
-							{ tab.title }
-						</TabButton>
-					) ) }
-				</NavigableMenu>
-				{ selectedTab && (
-					<div
-						aria-labelledby={ selectedId }
-						role="tabpanel"
-						id={ selectedId + '-view' }
-						className="components-tab-panel__tab-content"
+						) }
+						tabId={ instanceId + '-' + tab.name }
+						aria-controls={ instanceId + '-' + tab.name + '-view' }
+						selected={ tab.name === selected }
+						key={ tab.name }
+						onClick={ partial( handleClick, tab.name ) }
 					>
-						{ this.props.children( selectedTab ) }
-					</div>
-				) }
-			</div>
-		);
-	}
-}
+						{ tab.title }
+					</TabButton>
+				) ) }
+			</NavigableMenu>
+			{ selectedTab && (
+				<div
+					aria-labelledby={ selectedId }
+					role="tabpanel"
+					id={ selectedId + '-view' }
```

Comment:
Our coding guidelines [recommend](https://developer.wordpress.org/block-editor/contributors/develop/coding-guidelines/#strings) using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) instead of string concatenation whenever possible:
```suggestion
					id={ `${ selectedId }-view` }
```